confounds/metrics.py

```diff
@@ -6,3 +6,102 @@
 degree of harmonization achieved (e.g. reduction in variance of means/medians)
 
 """
+
+import numpy as np
+
+from confounds import Residualize
+from scipy import stats
+
+
+def partial_correlation(X, y=None):
+    """
+    Calculates the pairwise partial correlations between all of the variables in X with respect to some confounding variables y
+
+    Can be used as a measure of the strength of the relationship between two variables of interest while controlling for some other variables.
+
+    Parameters
+    ----------
+    X : {array-like, sparse matrix}, shape (n_samples, n_features)
+            The training input samples.
+    y : {array-like, sparse matrix}, shape (n_samples, n_covariates)
+            This does not refer to target as is typical in scikit-learn.
+
+    Returns
+    -------
+    partial_correlations : ndarray
+        Returns the pairwise partial correlations of each variable in X
+    """
+    resx = Residualize()
+    resx.fit(X, y)
+    deconfound_X = resx.transform(X, y)
+    return np.corrcoef(deconfound_X, rowvar=False)
+
+
+def partial_correlation_t_test(X, y=None):
+    """
+    Calculates the t-statistic and p-value for pairwise partial correlations between all of the variables in X with
+    respect to some confounding variables y.
+
```

Comment:
it would be useful to mention when and how this would be used! When we get a chance we can write a jupyter notebook demo later, but mentioning of few examples would be helpful.

Reply:
@jameschapman19 few sentences for this one?